File: BookReading.cpp
#include<iostream>
using namespace std;
const int N=1e5+10;

int main() {
	freopen("BookReading.in", "r",stdin);
    int t, n, m ,q, r;
    int page[N], ans[N];
    long long res = 0;
    scanf("%d", &t);
    for (int cas = 1; cas <= t; cas++) {
        scanf("%d%d%d", &n, &m, &q);
        for (int j = 1; j <= n;j++) {
            page[j] = 1;
        }
        int torn;
        for (int j = 1; j <= m; j++) {
            scanf("%d", &torn);
            page[torn] = 0;
        }
        
		// for (int j = 1; j <= n; j++) {
        //     for (int k = j; k <= n; k += j) {
        //         ans[j] += page[k];
        //     } 
        // }
        // for (int j = 0; j < q; j++){
        //     scanf("%d", &r);
        //     res += ans[r];
        // }

		for(int i=1;i<=n;i++) for(int j=i;j<=n;j += i) ans[i]+=page[j];
		long long sum=0;
		for(int i=1;i<=q;i++) scanf("%d",&r),sum+=ans[r];
		
		printf("Case #%d: %lld\n", cas, sum);
		for (int j = 1; j <= n;j++) {
            page[j] = 0;
			ans[j] = 0;
        }
    }
}
            

File: BookReading.py
# round G 2019 - Kick Start 2019
import sys
sys.stdin = open('BookReading.in', 'r')

# 算法复杂度 nlogn
def main():
    num = int(input())
    for i in range(num):
        M_N_Q = list(map(int, input().split(' ')))
        M = set(map(int, input().split(' ')))
        Q = list(map(int, input().split(' ')))
        res = 0
        for reader in range(M_N_Q[2]):
            for page in range(Q[reader], M_N_Q[0] + 1, Q[reader]):
                if page not in M:
                    res += 1
        print("Case #" + str(i + 1) + ":", res)

if __name__ =='__main__':
    main()